lib/module.js

```diff
@@ -632,6 +637,11 @@ Module._extensions['.node'] = function(module, filename) {
   return process.dlopen(module, path._makeLong(filename));
 };
 
+if (harmonyModules) {
+  Module._extensions['.mjs'] = function(module, filename) {
+    throw new errors.Error('ERR_REQUIRE_ESM', filename);
+  };
```

Comment:
Is throwing here a good idea for cjs? There will likely be tools that tap into that extensions hook, like `babel-register`, `flow-remove-types`, etc.